src/backend/gl/src/lib.rs

```diff
@@ -315,6 +316,162 @@ impl PhysicalDevice {
         let vendor: std::string::String = info.platform_name.vendor.clone();
         let renderer: std::string::String = info.platform_name.renderer.clone();
 
+        // An array of hal `MemoryType`s each corresponding to a set of memory types generated for
+        // every allowed role.  In the OpenGL backend, depending on platform capabilities, we may
+        // need multiple different memory types with the same properties / heap_index but different
+        // allowed opengl roles.
+        let mut buffer_role_memory_types = Vec::new();
+
+        // Mimicking vulkan, memory types with more flags should come before those with fewer flags
+        if private_caps.map {
+            buffer_role_memory_types.push(hal::MemoryType {
+                properties: memory::Properties::CPU_VISIBLE | memory::Properties::CPU_CACHED | memory::Properties::COHERENT,
+                heap_index: CPU_VISIBLE_HEAP,
+            });
+            buffer_role_memory_types.push(hal::MemoryType {
```

Comment:
are we missing the CPU-visible non-cached one?

Reply:
You know what, I actually did all this still with the assumption that there would only be R/W memory types this patch and not write-only ones (and leave that for a later patch).  I guess though, with the way everything else is set up, the ONLY THING left to do is just add an entry here and it will work?  I guess that's really all that's left to do so it's silly not to do it!